File: math_stuff/less_interesting_languages/triangle.js
"use strict";
// run with `node triangle.js`

// helper functions --------
function range(upperLimit) {
	return [...Array(upperLimit + 1).keys()];
}

function mapToRange(func, upperLimit) {
	return range(upperLimit).map(x => func(x));
}
// -------------------------

// functional(?) style
const triangle = number =>
	range(number)
		.reduce((num, acc) => num + acc);

// imperative style
function pyramid(number) {
	let result = 0;
	for (let i = 1; i <= number; i++)
		result += i * i;
	return result;
}

console.log("triangle numbers")
console.log(mapToRange(triangle, 10))

console.log("pyramid numbers")
console.log(mapToRange(pyramid, 10))

File: math_stuff/less_interesting_languages/triangle.c
// run with `gcc -o triangle.out triangle.c && ./triangle.out`

#include <stdio.h>

int triangle(int number) {
	int result = 0;
	for (int i = 1; i <= number; i++)
		result += i;
	return result;
}

int pyramid(int number) {
	int result = 0;
	for (int i = 1; i <= number; i++)
		result += i * i;
	return result;
}

int main() {
	printf("triangle numbers:\n");
	for (int i = 1; i <= 10; i++)
		printf("%d ", triangle(i));

	printf("\npyramid numbers:\n");
	for (int i = 1; i <= 10; i++)
		printf("%d ", pyramid(i));

	printf("\n");
}
